metals/src/main/scala/scala/meta/internal/metals/InteractiveSemanticdbs.scala

```diff
@@ -78,8 +79,8 @@ final class InteractiveSemanticdbs(
           if (existingDoc == null || existingDoc.md5 != sha) {
             val compiled = compile(path, text)
             Option(compiled).foreach(doc =>
-              // don't index dependency source files
-              if (!source.isDependencySource(workspace))
+              // don't index dependency source files or worksheets, since their definitions are local
+              if (!source.isDependencySource(workspace) && !source.isWorksheet)
```

Comment:
We don't want to index worksheets globally at least until we can distinguish different build targets and unrelated symbols with the same name. Also, worksheet definitions are only available inside the worksheet itself.